pkg/client/leaderelection/resourcelock/configmaplock.go

```diff
@@ -35,7 +35,7 @@ type ConfigMapLock struct {
 	// ConfigMapMeta should contain a Name and a Namespace of an
 	// ConfigMapMeta object that the Leadercmlector will attempt to lead.
 	ConfigMapMeta metav1.ObjectMeta
-	Client        clientset.Interface
+	Client        corev1client.ConfigMapsGetter
```

Comment:
Can we pass [corev1client.ConfigMapInterface](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/clientset_generated/clientset/typed/core/v1/configmap.go#L35) around, i.e., passing Client.ConfigMaps(ConfigMapMeta.Namespace) around? It looks like [federation.ConfigMapInterface](https://github.com/kubernetes/kubernetes/blob/master/federation/client/clientset_generated/federation_clientset/typed/core/v1/configmap.go#L35) has the same methods as corev1client.ConfigMapInterface, so you can reuse the leaderelection code for federation.